test/Go/GoListPackagesSpec.hs

```diff
@@ -0,0 +1,241 @@
+xSys =
+  Dependency
+    { dependencyType = GoType
+    , dependencyName = "golang.org/x/sys"
+    , dependencyVersion = Just (CEq "v0.0.0-20220114195835-da31bd327af9")
+    , dependencyLocations = []
+    , dependencyEnvironments = Set.singleton EnvProduction
+    , dependencyTags = Map.empty
+    }
+
+xxhash :: Dependency
+xxhash =
+  Dependency
+    { dependencyType = GoType
+    , dependencyName = "github.com/cespare/xxhash/v2"
+    , dependencyVersion = Just (CEq "v2.1.2")
+    , dependencyLocations = []
+    , dependencyEnvironments = Set.singleton EnvProduction
+    , dependencyTags = Map.empty
+    }
+
+protobufOld :: Dependency
+protobufOld =
+  Dependency
+    { dependencyType = GoType
+    , dependencyName = "google.golang.org/protobuf"
+    , dependencyVersion = Just (CEq "v1.26.0")
+    , dependencyLocations = []
+    , dependencyEnvironments = Set.singleton EnvProduction
+    , dependencyTags = Map.empty
+    }
+
+expected :: Graphing.Graphing Dependency
+expected = run . evalGrapher $ do
+  traverse_ direct [clientGolang, xdiff]
+  traverse_
+    deep
+    [ perks
+    , protobuf
+    , golangProtobufExtensions
+    , clientModel
+    , common
+    , procfs
+    , xSys
+    , xxhash
+    , protobufOld
+    ]
+
+  edges
+    [ (protobuf, protobufOld)
+    , (golangProtobufExtensions, protobuf)
+    , (clientModel, protobuf)
+    , (common, protobuf)
+    , (common, golangProtobufExtensions)
+    , (common, clientModel)
+    , (procfs, xSys)
+    , (clientGolang, perks)
+    , (clientGolang, protobuf)
+    , (clientGolang, protobufOld)
+    , (clientGolang, clientModel)
+    , (clientGolang, common)
+    , (clientGolang, procfs)
+    , (clientGolang, protobuf)
+    , (clientGolang, xxhash)
+    ]
+
+-- | This FilePath represents =go list -json -deps all= output with a local package replacement and transitive deps.
+testFile :: FilePath
+testFile = "test/Go/testdata/go-list-pkgs-out.json"
+
+ignoresC :: Spec
+ignoresC =
+  describe "Using C bindings" $
+    it "Ignores the special \"C\" package in dependencies" $ do
+      let result =
+            run . runStack . runDiagnostics $
+              buildGraph
+                [ GoPackage
+                    { importPath = ImportPath "github.com/ajankovic/xdiff"
+                    , standard = False
+                    , moduleInfo =
+                        Just
+                          GoModule
+                            { modulePath = ModulePath "github.com/ajankovic/xdiff"
+                            , version = Just (ModuleVersion "v0.0.1")
+                            , indirect = False
+                            , isMainModule = False
+                            , replacement = Nothing
+                            }
+                    , packageDeps = [ImportPath "C"]
+                    , listError = Nothing
+                    }
+                ]
+      assertOnSuccess result $ \_ (graph, _) -> graph `shouldBe` Graphing.direct xdiff
+
+analysisSpec :: Spec
```

Comment:
Can you add a test case here for 1 dep that requires a replacement and 1 dep that is a path dependency?

Reply:
I've replaced the unit tests and included this as a case in them.